transformation/src/main/scala/org/broadinstitute/monster/hca/HcaPipelineBuilder.scala

```diff
@@ -145,7 +145,30 @@ object HcaPipelineBuilder extends PipelineBuilder[Args] {
         Str("content") -> Str(encode(metadata)),
         Str("crc32c") -> Str(contentHash),
         Str("source_file_id") -> Str(fileId),
-        Str("source_file_version") -> Str(fileVersion)
+        Str("source_file_version") -> Str(fileVersion),
+        Str("data_file_name") -> Str(dataFileName)
+      )
+    )
+  }
+
+  /**
+    * Extract the necessary info from file metadata and put it into a form that
+    * can be used to generate bulk file ingest requests.
+    * @param metadata the content of the metadata file in Msg format
+    * @param inputPrefix the root directory containing input files
+    * @return a Msg object in the desired output format
+    */
+  def generateFileIngestRequest(metadata: Msg, inputPrefix: String): Msg = {
+    val contentHash = metadata.read[String]("crc32c")
+    val dataFileName = metadata.read[String]("data_file_name")
+    val sourcePath = s"$inputPrefix/data/$dataFileName"
+    val targetPath = s"/$dataFileName"
```

Comment:
Sanity-checking: Are we sure the data file names don't start with `/`?

Reply:
I just checked, and no, they don't start with `/`.